staging/src/k8s.io/kubectl/pkg/cmd/cp/cp.go

```diff
@@ -378,13 +329,13 @@ func recursiveTar(srcBase, srcFile, destBase, destFile string, tw *tar.Writer) e
 			if len(files) == 0 {
 				//case empty directory
 				hdr, _ := tar.FileInfoHeader(stat, fpath)
-				hdr.Name = destFile
+				hdr.Name = destFile.String()
 				if err := tw.WriteHeader(hdr); err != nil {
 					return err
 				}
 			}
 			for _, f := range files {
-				if err := recursiveTar(srcBase, path.Join(srcFile, f.Name()), destBase, path.Join(destFile, f.Name()), tw); err != nil {
+				if err := recursiveTar(srcBase, srcFile.Join(newLocalPath(f.Name())), destBase, destFile.Join(newRemotePath(f.Name())), tw); err != nil {
```

Comment:
How do you know that `src` is local and `dest` is remote here?

Reply:
oh, I think because this method is only ever called locally? Remote tar always execs the binary?

In that case, maybe the parameters should use the locol/remote types rather than the generic interface, to make this explicit?

Same goes for `untar` below.